src/Ztm.WebApi/TransactionConfirmationWatchers/IRuleRepository.cs

```diff
@@ -0,0 +1,32 @@
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using NBitcoin;
+using Ztm.WebApi.Callbacks;
+
+namespace Ztm.WebApi.TransactionConfirmationWatchers
+{
+    public interface IRuleRepository
+    {
+        Task<Rule> AddAsync
+        (
+            uint256 transaction,
+            int confirmation,
+            TimeSpan unconfirmedWaitingTime,
+            CallbackResult successResponse,
+            CallbackResult timeoutResponse,
+            Callback callback,
+            CancellationToken cancellationToken
+        );
+
+        Task ClearCurrentWatchIdAsync(Guid id, CancellationToken cancellationToken);
+        Task<Rule> GetAsync(Guid id, CancellationToken cancellationToken);
+        Task<TimeSpan> GetRemainingWaitingTimeAsync(Guid id, CancellationToken cancellationToken);
+        Task<RuleStatus> GetStatusAsync(Guid id, CancellationToken cancellationToken);
+        Task<IEnumerable<Rule>> ListWaitingAsync(CancellationToken cancellationToken);
```

Comment:
I think `ListNoCurrentWatchAsync` looks better.

Reply:
We have two filters for this
1) Current watch id is null.
2) Status of rule is pending.

Thanks for your idea but I believe the name is not clear enough.